### PR TITLE
Change how config reload works

### PR DESCRIPTION
**Pull request by clockworksoul**
Config no longer reloads whenever the config file changes.

Instead, it'll issue a reload under one of the two conditions:

* A `SIGHUP` signal to the Gort process
* A `GET` request to the `/v2/reload` endpoint

If a reload is requested but the file hasn't changed, no reload will occur.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -246,33 +229,38 @@ func loadConfiguration(file string) (*data.GortConfig, error) {
...
+// Reload is called by Initialize() to determine whether the config file has changed (or is new) and reload if it has.
+func Reload() error {
 	configMutex.Lock()
 	defer configMutex.Unlock()
 
 	sum, err := getMd5Sum(configFile)
 	if err != nil {
+		log.
+			WithField("file", configFile).
+			WithError(err).Error("Configuration reload failed")
```
Comment (theothertomelliott) [inline config/config.go:241]: Is there other context we can add to these error messages to make it clear where the failure happened?

File: gort.go [hunk 2]
```diff
@@ -76,6 +74,8 @@ func installAdapters() error {
 func startGort(ctx context.Context, configFile string, verboseCount int) error {
 	setLoggerVerbosity(verboseCount)
 
+	go catchSignals()
```
Comment (theothertomelliott) [inline gort.go:77]: What happens if SIGHUP is sent before initializeConfig returns?

**Comment by theothertomelliott** [comment]
Got a couple questions, but general concept LGTM!

File: gort.go (hunk 2 above)
Comment (clockworksoul) [reply gort.go:77]: Good thinking! 

Both operations find their way to `config.Reload`, which locks on `configMutex`. However, the specifics largely depend on whether `signal.Notify` has completed by the time the `SIGHUP` signal comes in.

* If it HAS NOT, then the `SIGHUP` will be handled according to default behavior, which will cause Gort to exit. 
* If it HAS, then the signal will cause `config.Reload` to be called, which will block on the mutex resulting in two calls to `Reload`. If the underlying config file hasn't changed (per checksum) then the second call will be ignored.
 
In other words, nothing unusual. :)

File: config/config.go (hunk 1 above)
Comment (clockworksoul) [reply config/config.go:241]: The message is just commentary, really. The text of the actual error proper is emitted by the `WithError`. I can standardize the comment message a bit though.

**Review by theothertomelliott — APPROVED** [review]
